src/vma/proto/rule_table_mgr.cpp

```diff
@@ -131,33 +131,45 @@ bool rule_table_mgr::parse_enrty(nlmsghdr *nl_header, rule_val *p_val)
 // Parameters: 
 //		rt_attribute	: object that contain rule attribute.
 //		p_val			: custom object that contain parsed rule data.
-void rule_table_mgr::parse_attr(struct rtattr *rt_attribute, rule_val *p_val)
+void rule_table_mgr::parse_attr(struct rtnl_rule *rule, rule_val *p_val)
 {
-	switch (rt_attribute->rta_type) {
-		case FRA_PRIORITY:
-			p_val->set_priority(*(uint32_t *)RTA_DATA(rt_attribute));
-			break;
-		case FRA_DST:
-			p_val->set_dst_addr(*(in_addr_t *)RTA_DATA(rt_attribute));
-			break;
-		case FRA_SRC:
-			p_val->set_src_addr(*(in_addr_t *)RTA_DATA(rt_attribute));
-			break;
-		case FRA_IFNAME:
-			p_val->set_iif_name((char *)RTA_DATA(rt_attribute));
-			break;
-		case FRA_TABLE:
-			p_val->set_table_id(*(uint32_t *)RTA_DATA(rt_attribute));
-			break;
+	// FRA_PRIORITY: Rule Priority
+    uint32_t priority = rtnl_rule_get_prio(rule);
+    if (priority) {
+        p_val->set_priority(priority);
+    }
+
+    // FRA_DST: Destination Address
+    struct nl_addr *dst = rtnl_rule_get_dst(rule);
+    if (dst) {
+        p_val->set_dst_addr(*(in_addr_t *)nl_addr_get_binary_addr(dst));
+    }
+
+    // FRA_SRC: Source Address
+    struct nl_addr *src = rtnl_rule_get_src(rule);
+    if (src) {
+        p_val->set_src_addr(*(in_addr_t *)nl_addr_get_binary_addr(src));
+    }
+
+    // FRA_IFNAME: Input Interface Name
+    char *iif_name = rtnl_rule_get_iif(rule);
+    if (iif_name) {
+        p_val->set_iif_name(iif_name);
+    }
+
+    // FRA_TABLE: Table ID
+    uint32_t table_id = rtnl_rule_get_table(rule);
+    if (table_id) {
+        p_val->set_table_id(table_id);
+    }
+
 #if DEFINED_FRA_OIFNAME
-		case FRA_OIFNAME:
-			p_val->set_oif_name((char *)RTA_DATA(rt_attribute));
-			break;
+    // FRA_OIFNAME: Output Interface Name (if available)
+    char *oif_name = rtnl_rule_get_oif(rule);
+    if (oif_name) {
+        p_val->set_oif_name(oif_name);
+    }
```

Comment:
please consider adapting your IDE for product format style.

Reply:
Thanks! didn't have clang format applied for this repo. not a native dev here :D
went through all the files and applied formatting